### ISSUE TITLE
Gradle wrapper in Java templates too far out of date

### ISSUE THREAD

**Issue by rli** [aws/aws-sam-cli-app-templates#87]
On systems where JVM 15 is the default, the GradleBuild workflow will always fail:

```
/usr/libexec/java_home -V
Matching Java Virtual Machines (3):
    15.0.2, x86_64:	"Amazon Corretto 15"	/Library/Java/JavaVirtualMachines/amazon-corretto-15.jdk/Contents/Home
    11.0.9.1, x86_64:	"Amazon Corretto 11"	/Library/Java/JavaVirtualMachines/amazon-corretto-11.jdk/Contents/Home
    1.8.0_275, x86_64:	"Amazon Corretto 8"	/Library/Java/JavaVirtualMachines/amazon-corretto-8.jdk/Contents/Home

/Library/Java/JavaVirtualMachines/amazon-corretto-15.jdk/Contents/Home
```

```
$ sam init --name j11 --runtime java11 --app-template hello-world --package-type Zip --dependency-manager gradle
$ cd j11
<j11> $ sam build --template template.yaml

Building codeuri: HelloWorldFunction runtime: java11 metadata: {} functions: ['HelloWorldFunction']
/Users/richali/tmp/sam/j11/HelloWorldFunction/gradlew is using a JVM with major version 15 which is newer than 11 that is supported by AWS Lambda. The compiled function code may not run in AWS Lambda unless the project has been configured to be compatible with Java 11 using 'targetCompatibility' in Gradle.
Running JavaGradleWorkflow:GradleBuild

Build Failed
Error: JavaGradleWorkflow:GradleBuild - Gradle Failed: FAILURE: Build failed with an exception.

* What went wrong:
Could not open cp_init remapped class cache for 8kkh2ehmk58t6twxmma1ubqof (/Users/richali/.gradle/caches/5.1.1/scripts-remapped/lambda_build_init_7epsn6nrbbjv0w9bodo48jyjo/8kkh2ehmk58t6twxmma1ubqof/cp_init13bba240a07c562d578336deed5b2fbf).
> Could not open cp_init generic class cache for initialization script '/var/folders/5s/kz556x054vz8l7204yh4j9krq6bwmm/T/tmpr934w_w0/lambda-build-init.gradle' (/Users/richali/.gradle/caches/5.1.1/scripts/8kkh2ehmk58t6twxmma1ubqof/cp_init/cp_init13bba240a07c562d578336deed5b2fbf).
   > Could not initialize class org.codehaus.groovy.vmplugin.v7.Java7

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```

Fix is to update the gradle wrapper:
```
<j11> $ (cd HelloWorldFunction && ./gradlew wrapper --gradle-version=6.8.2 --distribution-type=bin)

BUILD SUCCESSFUL in 3s
1 actionable task: 1 executed

<j11> $ sam build --template template.yaml
Building codeuri: HelloWorldFunction runtime: java11 metadata: {} functions: ['HelloWorldFunction']
/Users/richali/tmp/sam/j11/HelloWorldFunction/gradlew is using a JVM with major version 15 which is newer than 11 that is supported by AWS Lambda. The compiled function code may not run in AWS Lambda unless the project has been configured to be compatible with Java 11 using 'targetCompatibility' in Gradle.
Running JavaGradleWorkflow:GradleBuild
Running JavaGradleWorkflow:CopyArtifacts

Build Succeeded
```

Additionally, it's probably a good idea to have the `targetCompatibility` parameter defined by default in the templates, since at least on macOS the standard behavior is to use the latest JVM available

**Comment by michaelbrewer** [comment]
https://github.com/aws/aws-sam-cli-app-templates/pull/161

**Comment by michaelbrewer** [comment]
@rli - i have a fix for this in PR #161

**Comment by rli** [comment]
thanks!
